server/controller/urlController.js

```diff
@@ -66,7 +68,48 @@ const createUrl = async (req, res) => {
   }
 };
 
+const signup = async (req, res) => {
+  try {
+    const { username, email, password } = req.body;
+    const user = await userModel.findOne({ email: email });
+    const hashedPassword = await bcrypt.hash(password, 10);
```

Comment:
`const hashedPassword = await bcrypt.hash(password, 10);`
Move this code should inside else block because this block will execute even when user exists